tests/IceRpc.Tests/Transports/Slic/SlicTransportSslAuthenticationTests.cs

```diff
@@ -102,42 +84,34 @@ public async Task Ssl_server_connection_connect_fails_when_client_provides_untru
         var sut = provider.GetRequiredService<ClientServerMultiplexedConnection>();
         var listener = provider.GetRequiredService<IListener<IMultiplexedConnection>>();
 
-        // Start the TLS handshake by calling connect on the client and server connections and wait for the
-        // connection establishment.
+        // Start the TLS handshake.
         var clientConnectTask = sut.Client.ConnectAsync(default);
 
         // Act/Assert
         IMultiplexedConnection? serverConnection = null;
         Assert.That(
             async () =>
             {
-                // We accept two behaviors here:
-                // - the listener can internally kill the client connection if it's not valid (e.g.: Quic behavior)
-                // - the listener can return the connection but ConnectAsync fails (e.g.: Slic behavior)
-                using var cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(250));
-                (serverConnection, _) = await listener.AcceptAsync(cts.Token);
+                (serverConnection, _) = await listener.AcceptAsync(default);
                 await serverConnection.ConnectAsync(default);
             },
-            Throws.TypeOf<AuthenticationException>().Or.InstanceOf<OperationCanceledException>());
+            Throws.TypeOf<AuthenticationException>());
 
         Assert.That(
             async () =>
             {
-                // We accept two behaviors here:
-                // - the client connection fails with AuthenticationException when try to create a stream
-                //   (e.g.: Quic behavior)
-                // - the client connect operation fails with either TransportException (e.g: Slic behavior).
                 if (serverConnection is not null)
                 {
                     await serverConnection.DisposeAsync();
```

Comment:
If we need to dispose the serverConnection before awaiting the clientConnectTask, it would be good to add a comment explaining why.

Reply:
That is not required in the simplified version